pkg/server/skydb/access.go

```diff
@@ -126,3 +130,192 @@ func (acl RecordACL) Accessible(userinfo *UserInfo, level RecordACLLevel) bool {
 
 	return accessible
 }
+
+// FieldAccessMode is the intended access operation to be granted access
+type FieldAccessMode int
+
+const (
+	// ReadFieldAccessMode means the access mode is for reading
+	ReadFieldAccessMode FieldAccessMode = iota + 1
+
+	// WriteFieldAccessMode means the access mode is for writing
+	WriteFieldAccessMode
+
+	// DiscoverFieldAccessMode means the access mode is for discovery
+	DiscoverFieldAccessMode
+
+	// CompareFieldAccessMode means the access mode is for query
+	CompareFieldAccessMode
+)
+
+// FieldACL contains all field ACL rules for all record types
+type FieldACL struct {
+	wildcardRecordType FieldACLEntryList
+	recordTypes        map[string]FieldACLEntryList
+}
```

Comment:
We could have used a single dictionary. Using a single array works but that is not efficient IMO. Using a map to find the first matching record type is O(1). Using an array to find the first matching record type is O(n).